### PR TITLE
Redact sensitive headers when logging prelude

### LINKED ISSUE #414: Avoid logging sensitive headers
Currently the entire HTTP prelude is printed at debug level, including all headers

https://github.com/algesten/ureq/blob/a1d7009e24185d05e3e411acdcf2b55103858b5e/src/unit.rs#L410

It would be nice to avoid printing sensitive headers like `Authorization`

### PR DESCRIPTION
**Pull request by andrewhickman**
Closes #414

### CODE REVIEW

File: src/unit.rs [hunk 1]
```diff
@@ -378,42 +377,117 @@ fn send_prelude(unit: &Unit, stream: &mut Stream, redir: bool) -> io::Result<()>
...
     // other headers
     for header in &unit.headers {
         if !redir || !header.is_name("Authorization") {
             if let Some(v) = header.value() {
-                write!(prelude, "{}: {}\r\n", header.name(), v)?;
+                if is_header_sensitive(header) {
+                    prelude.write_sensitive_header(header.name(), v)?;
+                } else {
+                    prelude.write_header(header.name(), v)?;
+                }
```
Comment (algesten) [inline src/unit.rs:405]: Wondering about the design choice here of having the `is_header_sensitive` external to `PreludeBuilder` at the same time as as the builder has `write_sensitive_header`. Not saying it's wrong, just wondering if there was a thought there?

File: src/unit.rs [hunk 2]
```diff
@@ -378,42 +377,117 @@ fn send_prelude(unit: &Unit, stream: &mut Stream, redir: bool) -> io::Result<()>
...
             }
         }
     }
 
     // finish
-    write!(prelude, "\r\n")?;
+    prelude.finish()?;
 
-    debug!("writing prelude: {}", String::from_utf8_lossy(&prelude));
+    debug!("writing prelude: {:?}", prelude);
```
Comment (algesten) [inline src/unit.rs:413]: Hm hm… I think we maybe should use the `Display` trait here and let the `Debug` trait just dump the internal state of the `PreludeBuilder`.

**Review by algesten — CHANGES_REQUESTED** [review]
Hi and thanks for the PR!

Looks good! I had a couple of small questions/things.

File: src/unit.rs (hunk 2 above)
Comment (andrewhickman) [reply src/unit.rs]: Changed to use `Display` instead

File: src/unit.rs (hunk 1 above)
Comment (andrewhickman) [reply src/unit.rs:405]: I think the reason I didn't include that logic inside PreludeBuilder is I wanted to re-use the `Header::is_name` function, but not all the headers we add come from an actual `Header` struct

**Review by jsha — APPROVED** [review]
This generally looks good to me. @algesten do you want to take a second look?

**Review by algesten — APPROVED** [review]
I think this is good to go

**Pull request merged**